### ISSUE TITLE
Magicleap InitFailed panic

### ISSUE THREAD

**Issue by asajeffrey** [servo/servo#24035]
Reported by @xclaesse:
```
-22 09:13:37.672  1595     1 I NoTag   : WR FrameBuilderConfig { default_font_render_mode: Subpixel, dual_source_blending_is_supported: true, dual_source_blending_is_enabled: true, chase_primitive: Nothing, enable_picture_caching: false, testing: false, gpu_supports_fast_clears: false, gpu_supports_advanced_blend: false, advanced_blend_is_coherent: true, batch_lookback_count: 10, background_color: None }
08-22 09:13:37.680  1595     1 I NoTag   : using KHR
08-22 09:13:37.699  1595    41 W NoTag   : stop_parsing for XML5 not implemented, full speed ahead!
08-22 09:13:37.720  1595    42 W NoTag   : Trying to get an event-loop from an unknown BC group
08-22 09:13:37.725  1595     4 I NoTag   : __bionic_open_tzdata_path: ANDROID_DATA not set!
08-22 09:13:37.726  1595     4 I NoTag   : thread 'ScriptThread PipelineId { namespace_id: PipelineNamespaceId(1), index: PipelineIndex(1) }' panicked at 'called `Result::unwrap()` on an `Err` value: InitFailed', src/libcore/result.rs:1084:5
08-22 09:13:37.726  1595     4 I NoTag   : note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
08-22 09:14:02.531  1595     1 I NoTag   : load_uri: https://duckduckgo.com/?q=r
08-22 09:14:02.532  1595    42 W NoTag   : LoadUrl for unknow browsing context: TopLevelBrowsingContextId(BrowsingContextId { namespace_id: PipelineNamespaceId(0), index: BrowsingContextIndex(1) })
08-22 09:14:39.399  1595     1 I NoTag   : load_uri: https://duckduckgo.com/?q=youtube.com
08-22 09:14:39.399  1595    42 W NoTag   : LoadUrl for unknow browsing context: TopLevelBrowsingContextId(BrowsingContextId { namespace_id: PipelineNamespaceId(0), index: BrowsingContextIndex(1) })
08-22 09:14:58.197  1595    42 W NoTag   : Closing browsing context BrowsingContextId { namespace_id: PipelineNamespaceId(0), index: BrowsingContextIndex(1) } twice.
08-22 09:14:58.197  1595     4 I NoTag   : thread 'LayoutThread PipelineId { namespace_id: PipelineNamespaceId(1), index: PipelineIndex(1) }' panicked at 'called `Result::unwrap()` on an `Err` value: RecvError', src/libcore/result.rs:1084:5
08-22 09:14:58.297  1595    42 W NoTag   : Sending script exit message failed (io error: crossbeam-channel receiver closed).
08-22 09:14:58.397  1595    42 W NoTag   : Sending script exit message failed (io error: crossbeam-channel receiver closed).
08-22 09:15:08.207  1595     1 W NoTag   : Incomplete shutdown.
08-22 09:15:08.307  1595     1 W NoTag   : Incomplete shutdown.
```

**Comment by asajeffrey** [comment]
IRC chat: https://mozilla.logbot.info/servo-magicleap/20190822#c16561081

TL;dr it works on my end, sigh.

**Comment by jdm** [comment]
SpiderMonkey is failing to initialize: https://github.com/servo/rust-mozjs/blob/master/src/rust.rs#L187

I suspect the JIT thing.

**Comment by asajeffrey** [comment]
Yes, not sure why my device isn't affected though. It could be a thing where the app is asking for permissions (which I've already granted on my device) and panicking due to going ahead without them?

**Comment by mrobinson** [comment]
Magic Leap support has been removed, so I'll close this one.
